ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java

```diff
@@ -481,9 +486,16 @@ public boolean exitMaintenanceMode() {
         throw new IllegalStateException("Cluster is not in maintenance mode");
       }
       try {
-        helixAdmin.manuallyEnableMaintenanceMode(clusterName, false, maintenanceModeReason, null);
+        MaintenanceRecord record = getLastMaintenanceRecord();
+        if (record.operationType.equals(MaintenanceRecord.OPERATION_TYPE_ENTER) && record.reason.equals(
+            maintenanceModeReason)) {
+          helixAdmin.manuallyEnableMaintenanceMode(clusterName, false, maintenanceModeReason, null);
```

Comment:
nit: I think it's a little weird to use `.manuallyEnableMaintenanceMode` to disable MM. And I believe the maintenance mode reason isn't utilized when disabling the MM, right? Python docs: https://docs.google.com/document/d/13UVVnWHqY1qBd8KXSZnIQs4dIFLeviDKOa7wpZp7Mf8/edit#bookmark=id.xps9dnsnrjw

Reply:
The reason is put in the maintenance history even if this is a disable operation.  I agree it's a bit weird, but this is the only API for manipulating MM in java client unfortunately.